joomla/joomla-cms
# Sync columns in contact_details table, add missing defaults, clean up

Pull Request for issue https://github.com/joomla/joomla-cms/issues/13289
To help #7680 on mssql

### Summary of Changes

1. Add missing defaults for columns [all db]:
* xreference
* sortname1, sortname2, sortname3
* ~~metakey, metadesc~~ - column type TEXT can not have default value for mysql

2. Add missing default for sqlsrv only:
* checked_out
* created_by_alias
* published

3. Remove defaults because that columns can not have default '' for :
* name
* alias
* language # for postgresql only

4. Remove `DEFAULT NULL` or alone `NULL` from sql files because this is redundant.

5. Clean up for postgresql:
Replace `DEFAULT 0 NOT NULL` to `NOT NULL DEFAULT 0`

Info:
https://github.com/joomla/joomla-cms/blob/staging/installation/sql/postgresql/joomla.sql#L264
https://github.com/joomla/joomla-cms/blob/staging/installation/sql/mysql/joomla.sql#L287

### Testing Instructions
Install joomla with applied changes. 
Installation should pass on all databases. 
Test create contact.
